### PR TITLE
Inline struct support

### PR DESCRIPTION
**Pull request by jrdnull**
Fixes: https://github.com/Teamwork/kommentaar/issues/10

### CODE REVIEW

**Comment by jrdnull** [comment]
Test is passing and working for crmcore apart from the reference is wrong for object types, e.g:

https://gist.github.com/jrdnull/4c4aa543426e887fdac08831125f1304#file-swagger-yml-L1313

the pipeline property:
```
          pipeline:
            $ref: v1.ResourceIdentifierRequest
```

it should be `crmcore.ResourceIdentifierRequest`, not sure how to fix this atm though

File: docparse/jsonschema.go [hunk 1]
```diff
@@ -240,6 +241,26 @@ start:
...
+		p.Properties = map[string]*Schema{}
+		for _, f := range typ.Fields.List {
+			prop, err := fieldToSchema(prog, "", ref, f)
+			if err != nil {
+				return nil, fmt.Errorf("anon struct: %v", err)
+			}
+
+			propName := f.Names[0].Name
+			if f.Tag != nil {
+				jsonTag := parseRawTag(f.Tag.Value, "json")
```
Comment (Carpetsmoker) [inline docparse/jsonschema.go]: There is the `goutil.Tag` helper for this as well: https://github.com/Teamwork/utils/blob/master/goutil/goutil.go#L215 – I think that does the same?

File: docparse/jsonschema.go [hunk 2]
```diff
@@ -354,7 +368,7 @@ arrayStart:
 	// Simple identifier: "string", "myCustomType".
 	case *ast.Ident:
 
-		dbg("resolveArray: ident: %#v", typ.Name)
+		dbg("resolveArray: ident: %#v in %s", typ.Name, pkg)
```
Comment (Carpetsmoker) [inline docparse/jsonschema.go]: Very minor thing, but IMHO it's better to use `%#v` or `%q` in debug messages. That way things like trailing spaces, empty strings, etc. show up a bit clearer.

File: docparse/jsonschema.go [hunk 3]
```diff
@@ -445,6 +460,7 @@ func JSONSchemaType(t string) string {
 }
 
 func getTypeInfo(prog *Program, lookup, filePath string) (string, error) {
+	dbg("getTypeInfo: %s in %s", lookup, filePath)
```
Comment (Carpetsmoker) [inline docparse/jsonschema.go]: Same here

**Pull request merged**